### DISCUSSION TITLE
The Json parser removes the "value" part if empty

### DISCUSSION THREAD

**Discussion by alysonroger** [dcm4che/dcm4che#1020] [Q&A] [answered]
Hello,
When I parse a DICOM object that I receive as input from a HTTP request, if this object has DICOM tags with empty value (which is acceptable in the norm as they are of type 2) , the "value" part of the tag is lost during the conversion **json -> Attributes.java -> json**.
 
**To reproduce**:
-  I start from a DICOM object that has the following tag:

`"00404070":{"vr":"SQ","value":[{"00404071":{"vr":"SQ","value":[{"21000140":{"vr":"AE","value":[]}}]}}]}`

- Then I convert this json into an Attributes with:

```
Attributes input = new Attributes();
JSONReader inputReader = new JSONReader(Json.createParser(newByteArrayInputStream(inputJson.getBytes(StandardCharsets.UTF_8))));
inputReader.readDataset(input);
```

I update a DICOM field at this moment but it is not relevant for this bug. 

-  Then I convert it back into Json to send it with an HTTP request

Here the variable "output" is of type Attributes and just equal to the input with a small DICOM update.
```
Writer writer = new StringWriter();
try (JsonGenerator generator = Json.createGenerator(writer)) {
  JSONWriter jsonWriter = new JSONWriter(generator);
  jsonWriter.write(output);
} catch (Exception e) {
  throw new DICOM2JsonException("Could not init jsonWriter" + e);
}

String outputDataString = writer.toString();
```

-  Then I see in my logs that the content of the outputDataString for the tag 21000140 is missing "value":

`"00404070":{"vr":"SQ","value":[{"00404071":{"vr":"SQ","value":[{"21000140":{"vr":"AE"}}]}}]}`

**Environment**:
- I am using the following version with Java 1.8:
```
        <dependency>
            <groupId>org.dcm4che</groupId>
            <artifactId>dcm4che-core</artifactId>
            <version>5.23.2</version>
            <scope>test</scope>
        </dependency>
        <dependency>
            <groupId>org.dcm4che</groupId>
            <artifactId>dcm4che-json</artifactId>
            <version>5.23.2</version>
        </dependency>
```

 I did not find any documentation about a way to ignore this removal. 

I attached a full version of the input here: [input.txt](https://github.com/dcm4che/dcm4che/files/7040531/input.txt)

Thank you for your help!

Alyson

**Comment by gunterze** [comment, marked as answer]
The behavior is intended in compliance with  [DICOM PS 3.18, F.2.5 DICOM JSON Model Null Values](http://dicom.nema.org/medical/dicom/current/output/html/part18.html#sect_F.2.5):

> If an attribute is present in DICOM but empty (i.e., Value Length is 0), it shall be preserved in the DICOM JSON attribute object containing no "Value", "BulkDataURI" or "InlineBinary".
